### PR TITLE
Fix some CORS issues in HTTP Nowhere mode

### PR DESCRIPTION
**Pull request by cschanaj**
* Rewrite access-control-allow-origin to avoid the HTTP protocol in
HTTP Nowhere mode

This change make CORS issues like #14275 less likely to happen in HTTP Nowhere mode

### CODE REVIEW

**Comment by cschanaj** [comment]
ping @Hainish

File: chromium/background-scripts/background.js [hunk 1]
```diff
@@ -591,27 +592,47 @@ function onHeadersReceived(details) {
...
-        return {};
       }
+
+      if (details.responseHeaders[idx].name.match(/Access-Control-Allow-Origin/i)) {
+        // Existing access-control-allow-origin header found
+        const value = details.responseHeaders[idx].value;
+
+        // If HTTP protocol is used, change it to HTTPS
+        if (value.match(/http:/)) {
+          details.responseHeaders[idx].value = value.replace(/http:/g, "https:");
```
Comment (YegorIevlev) [inline chromium/background-scripts/background.js]: What if one of the URLs has `http:` in the middle?

**Comment by Hainish** [comment]
Thanks @cschanaj.  This sounds correct to me.  Merging.

**Comment by cschanaj** [comment]
@Hainish Thanks for merging this PR. We don't have any full release since April, is there is plan to make a release soon?

**Comment by Hainish** [comment]
True - I'll put it on the schedule for early next week. Thanks for noting it.

On June 6, 2018 6:59:51 PM PDT, Chan Chak Shing <notifications@github.com> wrote:
>@Hainish Thanks for merging this PR. We don't have any full release
>since April, is there is plan to make a release soon?
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/EFForg/https-everywhere/pull/15606#issuecomment-395268575

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Pull request merged**